Kudu.FunctionalTests/OneDeployTests.cs

```diff
@@ -0,0 +1,389 @@
+                    {
+                        if (async == "true")
+                        {
+                            DeployResult result;
+                            do
+                            {
+                                result = await appManager.DeploymentManager.GetResultAsync("latest");
+                                Assert.Equal(deployer, result.Deployer);
+                                await Task.Delay(TimeSpan.FromSeconds(2));
+                            } while (!new[] { DeployStatus.Failed, DeployStatus.Success }.Contains(result.Status));
+                        }
+                        else
+                        {
+                            Assert.Equal(HttpStatusCode.OK, response.StatusCode);
+                        }
+                        await AssertSuccessfulDeploymentByFilenames(appManager, new string[] { "app.war" });
+                    }
+                }
+
+            });
+        }
+
+        [Fact]
+        public Task EnsureIterativeDeployment()
+        {
+            return ApplicationManager.RunAsync("TestIterativeDeployment", async appManager =>
+            {
+                var file1 = CreateRandomTestFile();
+                var response1 = await DeployNonZippedArtifact(appManager, file1, new ZipDeployMetadata(), "war", null);
+                response1.EnsureSuccessStatusCode();
+
+                var file2 = CreateRandomTestFile();
+                var response2 = await DeployNonZippedArtifact(appManager, file2, new ZipDeployMetadata(), "static", "iterative.txt");
+                response2.EnsureSuccessStatusCode();
+
+                await AssertSuccessfulDeploymentByFilenames(appManager, new string[] { "app.war", "iterative.txt" });
+            });
+        }
+
+
+
+        private static async Task<HttpResponseMessage> DeployNonZippedArtifact(
+            ApplicationManager appManager,
+            TestFile file,
+            ZipDeployMetadata metadata,
+            string type = null,
+            string path = null,
+            string async = "false")
+        {
+            TestTracer.Trace("Deploying file");
+            using (var fileStream = CreateFileStream(file))
+            {
+                IList<KeyValuePair<string, string>> queryParams = new List<KeyValuePair<string, string>>();
+                if (!string.IsNullOrWhiteSpace(type))
+                {
+                    queryParams.Add(new KeyValuePair<string, string>("type", type));
+                }
+                if (!string.IsNullOrWhiteSpace(path))
+                {
+                    queryParams.Add(new KeyValuePair<string, string>("path", path));
+                }
+                if (async != "false")
+                {
+                    queryParams.Add(new KeyValuePair<string, string>("async", async));
+                }
+                return await appManager.OneDeployManager.PushDeployFromStream(fileStream, metadata, queryParams);
```

Comment:
Move this to a helper function and call the helper here and elsewhere:
`var queryParams = GetOneDeployQueryParams(type, path, async)`